cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/ObjectMappedTableDefinition.java

```diff
@@ -104,4 +115,18 @@ private void validateSchema(Schema schema) throws UnsupportedTypeException {
       }
     }
   }
+
+  // we want to include the key as a column in the table for exploration, so add it to the schema
+  private Schema addKeyToSchema(Schema schema, String keyName, Schema.Type keyType) {
+    List<Schema.Field> fields = Lists.newArrayListWithCapacity(schema.getFields().size() + 1);
+    fields.add(Schema.Field.of(keyName, Schema.of(keyType)));
+    for (Schema.Field objectField : schema.getFields()) {
+      // have to lowercase since Hive will lowercase
+      if (keyName.toLowerCase().equals(objectField.getName().toLowerCase())) {
+        throw new IllegalArgumentException("Object field '" + keyName + "' cannot be named the same as the row key.");
```

Comment:
I would phrase the other way round. The row key cannot use the same column name as the object field.